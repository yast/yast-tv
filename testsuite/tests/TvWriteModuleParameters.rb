# encoding: utf-8

module Yast
  class TvWriteModuleParametersClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      # Read() return nil
      @READ_nil = { "modules" => { "options" => {} } }

      # Dir() returns current state of bttv
      @READ_one = {
        "modules" => {
          "options" => {
            "bttv" => [
              3,
              {
                "card"  => { 0 => nil, 1 => "3", 2 => "2" },
                "tuner" => { 0 => "a", 1 => "b", 2 => "c" }
              }
            ]
          }
        }
      }

      # Write() fails
      @WRITE_false = { "modules" => { "options" => false } }

      # Write() is OK
      @WRITE_true = { "modules" => { "options" => true } }

      # parameters which should be written
      @modules_pars = {
        "bttv"    => [
          3,
          {
            "card"  => { 0 => nil, 1 => "3", 2 => "2" },
            "tuner" => { 0 => "a", 1 => "b", 2 => "c" }
          }
        ],
        "isa-pnp" => [1, { "isapnp_reset" => { 0 => "0" } }]
      }


      TEST(lambda { Tv.WriteModulesParameters(@modules_pars) }, [
        @READ_nil,
        @WRITE_false,
        {}
      ], false)

      TEST(lambda { Tv.WriteModulesParameters(@modules_pars) }, [
        @READ_nil,
        @WRITE_true,
        {}
      ], false)

      # bttv has nil parameters -> nil will be written after calling Dir()
      @modules_pars = {
        "isa-pnp" => [1, { "isapnp_reset" => { 0 => "0" } }],
        "bttv" =>
          #		[ 3, $[ nil: $[] ]]
          [3, nil]
      }

      TEST(lambda { Tv.WriteModulesParameters(@modules_pars) }, [
        @READ_one,
        @WRITE_true,
        {}
      ], false)

      nil
    end
  end
end

Yast::TvWriteModuleParametersClient.new.main
