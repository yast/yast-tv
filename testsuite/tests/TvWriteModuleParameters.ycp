{

    include "testsuite.ycp";

    map READ = $[
	"etc" : $[ "install_inf" : $[] ],
	"target" : $[ "ycp" : $[] ],
	"probe" : $[ "system" : []],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    // Read() return nil
    map READ_nil =
	$[ "modules" :
	    $[ "options" : $[] ]
	];

	// Dir() returns current state of bttv
    map READ_one =
	$[ "modules" :
	    $[ "options" :
			$[ "bttv":
				[ 3, $[ "card": $[ 0: nil, 1: "3", 2: "2" ],
						"tuner": $[ 0: "a", 1: "b", 2: "c" ]] ]
			]]
	];

	// Write() fails
	map WRITE_false =
	$[ "modules" :
	    $[ "options" : false ]
	];

	// Write() is OK
	map WRITE_true =
	$[ "modules" :
	    $[ "options" : true ]
	];

	// parameters which should be written
	map modules_pars =
	$[ "bttv":
		[ 3, $[ "card": $[ 0: nil, 1: "3", 2: "2" ],
				"tuner": $[ 0: "a", 1: "b", 2: "c" ]]
		],
	   "isa-pnp":
		[ 1, $[ "isapnp_reset": $[ 0:"0"]]
		]
	];


    TEST(``(Tv::WriteModulesParameters (modules_pars)), [ READ_nil, WRITE_false, $[] ],false);

    TEST(``(Tv::WriteModulesParameters (modules_pars)), [ READ_nil, WRITE_true, $[] ], false);

	// bttv has nil parameters -> nil will be written after calling Dir()
	modules_pars =
	$[ "isa-pnp":
		[ 1, $[ "isapnp_reset": $[ 0:"0"]]
		],
	   "bttv" :
		[ 3, $[ nil: $[] ]]
	];

    TEST(``(Tv::WriteModulesParameters (modules_pars)), [ READ_one, WRITE_true, $[] ], false);
}
