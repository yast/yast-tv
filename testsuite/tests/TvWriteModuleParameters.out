Read	.target.ycp ["/var/lib/YaST2/installmap.ycp", $[]] $[]
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Write	.modules.options $["card":"3,2", "tuner":"a,b,c"] "bttv" false
Return	false
Write	.modules.options $["card":"3,2", "tuner":"a,b,c"] "bttv" true
Write	.modules.options $["isapnp_reset":"0"] "isa-pnp" true
Return	true
Dir	.modules.options: ["bttv"]
Write	.modules.options nil "bttv" true
Write	.modules.options $["isapnp_reset":"0"] "isa-pnp" true
Return	true
